impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtBuilder.java

```diff
@@ -295,10 +294,8 @@ public JwtBuilder claim(String name, Object value) {
     public String compact() {
 
         if (this.serializer == null) {
-            //try to find one based on the runtime environment:
-            InstanceLocator<Serializer<Map<String,?>>> locator =
-                Classes.newInstance("io.jsonwebtoken.impl.io.RuntimeClasspathSerializerLocator");
-            this.serializer = locator.getInstance();
+            // try to find one based on the services available
+            this.serializer = Services.loadFirst(Serializer.class);
```

Comment:
The only concern I have about this is that the runtime exception type will have changed since the previous implementation, and if anyone is catching that, their current code will not function correctly.  Perhaps we should wrap this exception with the previous one to guarantee compatibility?

Reply:
P.S. with a TODO statement to remove the wrapping for 1.0